userspace/libsinsp/plugin.cpp

```diff
@@ -50,6 +50,23 @@ using namespace std;
 
 static std::set<uint16_t> s_all_plugin_event_types = {PPME_PLUGINEVENT_E};
 
+static bool check_is_index(std::string& str)
+{
+	int length = str.length();
+	if(length == 0 || (length > 1 && str[0] == '0'))
+	{
+		return false;
+	}
+	for(int j = 0; j < length; j++)
+	{
+		if(!isdigit(str[j]))
```

Comment:
What if we use `strtol` here and directly return the number? This way, we avoid to duplicate this (below we call this function and then we call std::stoul(m_argstr);.
We could rename this one like:
`get_arg_idx()` that throws an exception on error.

Reply:
I created two different functions `extract_arg_index` and `extract_arg_key` to maintain a similar interface for both types of argument. I also added some comments in `extract_arg_index` to clarify a little bit the various steps. Please note that `std::stoul` alone is not enough as I wrote in the comment. We need further checks to be sure our indexes are valid

Reply:
Thanks for the kind explanation!

Reply:
Btw you can actually check stoul end pos, and check against your string pointer (ie: if it matched all string or just a part); that was what i meant!
In any case, we don't really need that optimization as this is only done during init phase.

Reply:
Yeah, the problem here is a little bit more complex because `stoul` recognizes as valid numbers also things like `-45`, `+45` `3.8` so I prefer to use a solid check in front of it. Btw, I agree with you, here we don't need to find a complex solution since we are in `init phase`.